tools/flaky-test-retryer/handler.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"log"
+
+	// TODO: remove this import once "k8s.io/test-infra" import problems are fixed
+	// https://github.com/test-infra/test-infra/issues/912
+	"github.com/knative/test-infra/tools/monitoring/prowapi"
+)
+
+// TODO(TrevorFarrelly): supported repos depend on flaky-test-reporter's
+// configuration. Use the reporter's config.yaml to build this list.
+var repos = []string{"serving"}
+
+// expectedMsg checks that the message we received is what we want to process
+func expectedMsg(msg *prowapi.ReportMessage) bool {
+	expStatus := msg.Status == prowapi.FailureState
+	expType := msg.JobType == prowapi.PresubmitJob
+
+	var expRepo bool
+	if len(msg.Refs) > 0 {
+		for _, repo := range repos {
+			if msg.Refs[0].Repo == repo {
```

Comment:
should this be Refs.Pulls[0].Org?

Reply:
we want to make sure the messages are coming from a repo we are collecting flaky results on, i.e. serving. It should be safe to assume that all of the messages we receive come from the knative org, so I'm not checking it at the moment.